JASP-Engine/JASP/R/commonbayesianttest.R

```diff
@@ -0,0 +1,1786 @@
+#
+# Copyright (C) 2013-2018 University of Amsterdam
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+.ttestBayesianReadData <- function(dataset = NULL,
+                                   dependents, grouping = NULL,
+                                   missing = c("excludeListwise", "excludeAnalysisByAnalysis")) {
+
+  missing <- match.arg(missing)
+
+  if (is.list(dependents)) {
+    dependents = unique(unlist(dependents))
+    dependents <- dependents[dependents != ""]
+  }
+  if (identical(grouping, ""))
+    grouping <- NULL
+
+  if (is.null(dataset)) {
+    excl <- grouping
+    if (missing == "excludeListwise") {
+      excl <- c(excl, dependents)
+    }
+    dataset <- .readDataSetToEnd(columns.as.numeric=dependents, columns.as.factor=grouping, exclude.na.listwise=excl)
+  }
+  return(dataset)
+}
+
+.ttestBayesianGetErrorsPerVariable <- function(options, dataset) {
+
+	if (options[["canDoAnalysis"]]) {
```

Comment:
I agree with Tim here, let's not make things more confusing for naive analysis-writers.

Reply:
How about the approach that @vankesteren used in the Binomial test? For example:

```r
if (!is.null(errors) && errors == "No variables") return()
```
or perhaps
```r
if (errors$noVariables) return()
```

Reply:
Kinda depends if we also want to be able to propogate normal errors that we'd actually display on top of a table/plot. Because we would never report this "error" as such, but instead display an empty table, so maybe a bit of a misnomer?

Some analyses use 
```
status <- list(
   ready = FALSE,
   errors = FALSE,
   errorMessage = ""
)
```
which makes a distinction between real errors we display and not being ready to run because we require more information. 

Not sure what the best option is, but we should be consistent.